erdpy/cli_delagation.py

```diff
@@ -0,0 +1,251 @@
+import sys
+from typing import Any
+
+from erdpy import cli_shared, utils
+from erdpy.delegation import staking_provider
+from erdpy.proxy import ElrondProxy
+from erdpy.transactions import do_prepare_transaction
+
+
+def setup_parser(subparsers: Any) -> Any:
```

Comment:
Not for this PR: 
We have to change the architecture in the following way:
1. we should avoid using arguments of type Any as that will make the debugging work more tedious
2. we should avoid as much as we can having some random placed functions and instead use methods: those can be better tested and replaced through polymorphism
3. I think this could have been a class, receiving all required components such as the proxy instance and such. This change in paradigm will make the code more testable and cleaner.